public/components/datasources/utils/helpers.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { S3GlueProperties } from 'common/types/data_connections';
+import { DatasourceDetails } from '../components/manage/data_connection';
+
+export function checkIsConnectionWithLakeFormation({
+  connector,
+  properties,
+}: DatasourceDetails): boolean {
+  return connector === 'S3GLUE' && (properties as S3GlueProperties)['glue.lakeformation.enabled'];
```

Comment:
@amsiglan Why have we kept lake formation as a sub-type of S3GLUE. Ideally it should have been kept as a separate datasource similar to how we have prometheus? Maybe someone from the backend team answer here. The basis of this question comes from: 

1. Can any S3 datasource be enabled with lake formation sub-type and be disabled back? If so what is the workflow on UI for this? If not, then we should have this as a separate type. 

2. From an implementation side can we do this logical abstraction on the main datasources page and then pass it down or maintain a global state?  Currently, we are calling this function in multiple pages to determine which type of datasource is it, which leads to multiple if/else conditions everywhere. If tomorrow, there's a new subtype will end up adding more if/else conditions everywhere. Let's try a more extensible way to add this subtype.

Reply:
It is a sub type because, lake formation is a kind of S3 GLUE connection and majority of the workflows are exactly same except for some cases for which we have diverged in the UI. Also, in the long term we will be working on further reducing the differences as much as possible.